source/GameInterface/Services/ItemRosters/Messages/Events/ItemRosterUpdated.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Common.Messaging;
+
+namespace GameInterface.Services.ItemRosters.Messages.Events
+{
+    /// <summary>
+    /// Called when an item roster is updated.
+    /// </summary>
+    public class ItemRosterUpdated : IEvent
+    {
+        public string PartyBaseId { get; }
+        public byte[] EquipmentElement { get; }
+        public int Number { get; }
+
+        public ItemRosterUpdated(string partyBaseId, byte[] equipmentElement, int number)
+        {
+            PartyBaseId = partyBaseId;
+            EquipmentElement = equipmentElement;
```

Comment:
It might be faster to create a protobuf serializer for EquipmentElement. see https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/Coop.Core/Surrogates/Vec2Surrogate.cs

Reply:
It won't improve serialization speed, it will just move it from one place to another. I need to look into minimizing the size of the object to be serialized.

Reply:
> It won't improve serialization speed, it will just move it from one place to another. I need to look into minimizing the size of the object to be serialized.

Actually protobuf serializers seems to be faster than the one we are using now. But i don't know how i should approach this.